include/git2/index.h

```diff
@@ -86,7 +86,7 @@ typedef struct git_index_entry {
 typedef enum {
 	GIT_IDXENTRY_EXTENDED  = (0x4000),
 	GIT_IDXENTRY_VALID     = (0x8000),
-} git_indxentry_flag_t;
+} git_idxentry_flag_t;
```

Comment:
Huh. This very much does look breaking to me?

Reply:
Sorry, a bit more constructive: what about also providing a second typedef? `typedef git_indxentry_flag_t git_idxentry_flag_t` so that nobody needs to change his code?

Reply:
Maybe, but wouldn't that introduce warnings in user code anyway because of the mismatch ? Thinking about it, I don't think it will, because the usage seems to be limited to bitwise operations. I'm just concerned that a `-Werror` downstream will have to change the code anyways.

Personally, I'd go for `git_index_entry_flag_t` for consistency, as that's the name of the struct, and it's part of the `index` system. I'm fine with just dropping the "n" though, since that's worst 😉.

Reply:
Sorry, yes, this was very much a WIP that is not at all ready for review.  Changing the title to reflect that.

Reply:
As far as the naming goes, I too would prefer `git_index_entry_...` except that the rest of the things in this file are named `git_idxentry`.  Possible that we can get away without breakage here, possibly not.

Appreciate the thoughts about `-Werror`.  I'm not sure that we've ever said that we guarantee that people can avoid warnings using libgit2 (and, indeed, you can see that the 32 bit platforms have loads of them that we should deal with but haven't yet).  But as that's something to strive for it's something interesting to keep in mind.